python/ribasim/ribasim/geometry/edge.py

```diff
@@ -112,13 +123,46 @@ def add(
             index=pd.Index([edge_id], name="edge_id"),
         )
 
+        self._validate_edge(to_node, from_node, edge_type)
```

Comment:
Can this line go up before creating the `table_to_append` and maybe even further? You only seem to need to_node, from_node, edge_type.